lib/mix/tasks/scenic.gen.component.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Mix.Tasks.Scenic.Gen.Component do
+  @shortdoc "Creates scaffolding for a new Scenic component."
+
+  @moduledoc """
+  Generates Scene boilerplate for a new component.
+
+  ```bash
+  mix scenic.gen.component component_name
+  ```
+  """
+  use Mix.Task
+
+  import Mix.Generator
+  alias ScenicNew.Common
+
+  @switches [
+    app: :string,
+    module: :string
+  ]
+
+
+  # --------------------------------------------------------
+  @doc false
+  def run(argv) do
+    {opts, argv} = OptionParser.parse!(argv, strict: @switches)
+
+    case argv do
+      [] ->
+        Mix.Tasks.Help.run(["scenic.gen.component"])
+
+      [component_name | _] ->
+
+        Common.elixir_version_check!()
+        app = opts[:app] || Path.basename(Path.expand(component_name))
+        Common.check_application_name!(app, !opts[:app])
+        mod = opts[:module] || Macro.camelize(app)
+        Common.check_mod_name_validity!(mod)
+        Common.check_mod_name_availability!(mod)
+
+        generate(app, mod)
+    end
+  end
+
+  # --------------------------------------------------------
+  defp generate(component_filename, component_module) do
+    if Mix.Project.umbrella?() do
+      Mix.raise "mix scenic.gen.component must be invoked from within your scenic application root directory."
+    end
```

Comment:
I believe that this should be on the top of `run/1` function. Firstly everything made before does not makes sense if we want to raise. Secondly it would be inconsistent and therefore confusing to display for example `help` message and when passing correct arguments `raise`.

Reply:
Fixed!